### PR TITLE
Fix type mismatch in Reader.readIntoBoundedBytes (#16416)

### LINKED ISSUE #16416: reader.readIntoBoundedBytes bug
### Zig Version

0.11.0-dev.4002+7dd1cf26f

### Steps to Reproduce and Observed Behavior

In this snippet:

https://github.com/ziglang/zig/blob/7dd1cf26f9b0cb104ada166a10ff356ca272577a/lib/std/io/reader.zig#L262-L272

line 268 `bytes_read` is type `Error!usize`. Line 270 adds that `usize` to an `IntFittingRange`:

https://github.com/ziglang/zig/blob/7dd1cf26f9b0cb104ada166a10ff356ca272577a/lib/std/bounded_array.zig#L35-L45

So if you create any bounded array say with size `10000`, `bounded.len` will have a type of `u17`, which the compiler correctly blocks and complains:

```
error: expected type 'u17', found 'usize'
note: unsigned 17-bit int cannot represent all possible unsigned 64-bit values
```

### Expected Behavior

`std.BoundedArrayAligned.Len` should be a `usize`, as opposed to `IntFittingRange`, just like all the other sizes in `std` (in this case the `read` call that returns a usize is the problem).

While the bounded array impl is cool because it only gets the smallest size necessary, the smol savings gains have *fun* side effects like this.

While the size storing of a min int is sus, its more sus that the reader is manipulating the slice and len directly in `readIntoBUfferedBytes` i can fix whichever one is decided is the buggy behavior if you want. but either way one of those should get changed

### PR DESCRIPTION
**Pull request by lockbox**
these changes address #16416 

Closes #16416 

- added typecast (that is guaranteed to be safe)
- added behavior test bug 16416 to test suite, all tests pass

Think i might actually be one of the only people who use this api but this should fix it now :')

### CODE REVIEW

File: test/behavior/bugs/16416.zig [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const std = @import("std");
```
Comment (rofrol) [inline test/behavior/bugs/16416.zig:1]: use descriptive file names instead of github issue number

Comment (lockbox) [reply test/behavior/bugs/16416.zig:1]: Is that something new? The convention for the behavior tests looks like the issue number should be the name of the file.

**Comment by lockbox** [comment]
looking at it again i need to move the behavior test from behavior tests into unit tests -- since there are currently none for these api's and they would have caught this compile time bug

File: test/behavior/bugs/16416.zig (hunk 1 above)
Comment (wooster0) [reply test/behavior/bugs/16416.zig:1]: Yes, please don't add any new files to test/behavior/bugs. The codebase shouldn't be tied to GitHub. Also, the tests should probably go at the bottom of lib/std/io/reader.zig with the rest of the tests.

Comment (ifreund) [reply test/behavior/bugs/16416.zig:1]: Indeed, these tests belong in the standard library source code. The `test/` directory is for compiler tests.

**Comment by lockbox** [comment]
updated branch as per my last comment, ready for review.

**Comment by andrewrk** [comment]
Thanks!

**Pull request merged**